File: vanvyapaar-frontend/src/services/buyerService.ts
import api from '../lib/api'
import { CartItem, Order, Product, Buyer, Review } from '../types'

export const buyerService = {
  // Cart operations
  addToCart: async (buyerId: number, productId: number, quantity: number = 1): Promise<CartItem> => {
    const response = await api.post(`/buyer/${buyerId}/cart/add/${productId}?quantity=${quantity}`)
    return response.data
  },

  getCart: async (buyerId: number): Promise<CartItem[]> => {
    const response = await api.get(`/buyer/${buyerId}/cart`)
    return response.data
  },

  updateCartItem: async (cartItemId: number, quantity: number): Promise<CartItem> => {
    const response = await api.put(`/buyer/cart/${cartItemId}?quantity=${quantity}`)
    return response.data
  },

  removeCartItem: async (cartItemId: number): Promise<void> => {
    await api.delete(`/buyer/cart/${cartItemId}`)
  },

  // Order operations
  placeOrder: async (buyerId: number): Promise<Order[]> => {
    const response = await api.post(`/buyer/${buyerId}/orders`)
    return response.data
  },

  getOrders: async (buyerId: number): Promise<Order[]> => {
    const response = await api.get(`/buyer/${buyerId}/orders`)
    return response.data
  },

  // Wishlist operations
  addToWishlist: async (buyerId: number, productId: number): Promise<Product> => {
    const response = await api.post(`/buyer/${buyerId}/wishlist/${productId}`)
    return response.data
  },

  getWishlist: async (buyerId: number): Promise<Product[]> => {
    const response = await api.get(`/buyer/${buyerId}/wishlist`)
    return response.data
  },

  removeFromWishlist: async (buyerId: number, productId: number): Promise<void> => {
    await api.delete(`/buyer/${buyerId}/wishlist/${productId}`)
  },

  // Profile operations
  getProfile: async (buyerId: number): Promise<Buyer> => {
    const response = await api.get(`/buyer/${buyerId}/profile`)
    return response.data
  },

  updateProfile: async (buyerId: number, profile: Partial<Buyer>): Promise<Buyer> => {
    const response = await api.put(`/buyer/${buyerId}/profile`, profile)
    return response.data
  },

  // Review operations
  addReview: async (buyerId: number, productId: number, rating: number, comment?: string): Promise<Review> => {
    const response = await api.post('/buyer/reviews', {
      buyerId,
      productId,
      rating,
      comment
    })
    return response.data
  },
}

export default buyerService

File: vanpaayaar-backend/src/main/java/com/tribal/service/BuyerService.java
package com.tribal.service;

import com.tribal.model.Cart;
import com.tribal.model.Order;
import com.tribal.model.Product;
import com.tribal.model.Buyer;

import java.util.List;

public interface BuyerService {
    // Products
    List<Product> listProducts();
    Product getProduct(Long productId);

    // Cart
    Cart addToCart(Long buyerId, Long productId, int quantity);
    List<Cart> getCart(Long buyerId);
    Cart updateCartItem(Long cartItemId, int quantity);
    void removeCartItem(Long cartItemId);

    // Orders
    List<Order> placeOrder(Long buyerId);
    List<Order> getOrders(Long buyerId);

    // Profile
    Buyer getProfile(Long buyerId);
    Buyer updateProfile(Long buyerId, Buyer updatedBuyer);

    // Search / Filter
    List<Product> searchProducts(String keyword);
    List<Product> filterProductsByCategory(String category);
    List<Product> filterProductsByPriceRange(double min, double max);
}
